include/smack/SmackRep.h

```diff
@@ -150,6 +166,14 @@ class SmackRep {
   virtual const Stmt* alloca(llvm::AllocaInst& i);
   virtual const Stmt* memcpy(const llvm::MemCpyInst& msi);
   virtual const Stmt* memset(const llvm::MemSetInst& msi);
+  virtual const Stmt* loadAsBytes(const llvm::LoadInst& li);
+  virtual const Stmt* storeAsBytes(const llvm::StoreInst& si);
```

Comment:
It is important to us to keep clean design and eliminate redundancy. But if I correctly understand your concern, then you could chain together the following methods
````
llvm::BasicBlock* llvm::Instruction::getParent()
llvm::Function* llvm::BasicBlock::getParent()
````
to get the function `I->getParent()->getParent()` from an instruction pointer I.